assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -18,14 +21,24 @@ import SavedPaymentMethodOptions from './saved-payment-method-options';
  */
 const PaymentMethods = () => {
 	const { isInitialized, paymentMethods } = usePaymentMethods();
-	const [ showNewPaymentMethods, setShowNewPaymentMethods ] = useState(
-		true
-	);
-	const onChange = useCallback(
+	const [ savedMethodsToken, setSavedMethodsToken ] = useState( '' );
+	const [ paymentMethodsName, setPaymentMethodsName ] = useState( '' );
```

Comment:
I don't think you'll need state for these. The payment method context should handle all this (and provide it to the component). Take a look at how `usePaymentMethodDataContext` and `usePaymentMethodInterface` hooks are used in `trunk` to get and change the active payment method.

Reply:
@budzanowski I ended up removing lots of this when I moved saved payment method token (saved card) state into context. 

I don't think this caused problems, but there may be more tidying we can do here. There's some preexisting local state in `SavedPaymentMethodOptions` which I don't fully understand. We might be able to move that logic to context too, since the active payment method / token are closely related (i.e. same set of radio buttons).